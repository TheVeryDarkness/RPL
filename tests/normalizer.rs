#![cfg(test)]

use std::fs::read_to_string;
use std::path::Path;

use ui_test::per_test_config::TestConfig;
use ui_test::{Config, Match, error_on_output_conflict};

#[test]
fn normalize_stderr() {
    let mut cfg = Config::rustc(env!("CARGO_MANIFEST_DIR"));
    cfg.comment_defaults
        .base()
        .normalize_stderr
        .push((Match::PathBackslash, b"/".to_vec()));
    let stderr_path = Path::new("tests/normalizer.stderr");
    let test_path = Path::new("tests/normalizer.rs");
    let mut text = read_to_string(stderr_path).unwrap();
    assert!(text.contains('/'));
    text = text.replace("/", "\\");
    let mut errors = Vec::new();
    let test_cfg = TestConfig::one_off_runner(cfg, test_path.to_path_buf());
    error_on_output_conflict(stderr_path, text.as_bytes(), &mut errors, &test_cfg);
    assert!(
        errors.is_empty(),
        "Expected no errors, got: {:?}",
        errors.first().unwrap()
    );
}
