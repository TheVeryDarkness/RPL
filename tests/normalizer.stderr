error: transmute from `std::vec::Vec<u16>` to `std::vec::Vec<u32>` with mismatched layout is unsound
  --> tests/ui/ub/unsound_collection_transmute/lib.rs:4:14
   |
LL |     unsafe { transmute::<_, Vec<u32>>(vec![2_u16]) };
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^-----------^
   |                                       |
   |                                       casted from this
   |
   = note: `#[deny(rpl::unsound_collection_transmute)]` on by default